### DISCUSSION TITLE
ml/consciousness

### DISCUSSION THREAD

**Discussion by joe-antognini** [joe-antognini/joe-antognini.github.io#27] [Comments]
Written on 06/17/2022 19:13:39 

URL: https://joe-antognini.github.io/ml/consciousness

**Comment by joe-antognini** [comment]
Comment written by **Mixy Pisa** on **06/20/2022 08:25:44**

<blockquote>To say that a physical system is a computer requires an external observer to map the physical states of that system onto the abstract states of a Turing machine.</blockquote><p>That is only to <i>say </i>that a physical system is a computer.  The physical system is a computer whether you say so or not.</p>

**Comment by joe-antognini** [comment]
Comment written by **Jack Faller** on **06/20/2022 09:26:02**

<p>You seem to misunderstand the concept of Turing machines in your iron example. A computer can be understood as a category whose objects are states and whose morphisms are the progressions between states. For a Turing machine T and your iron I, what makes I a sort of Turing machine is a functor from T to I (this shows that a computer is more than a set of states—on some level, it is the relationships between states). The functor you propose is from I to T, which doesn't satisfy this. What makes something a computer is not that it can be mapped to any other computer, but that any other computer can be mapped to it. (edit) To clarify, this is really just saying that they aught to be isomorphic. You provide a one way mapping where a pair of functors should be needed. It should also be noted that the nature of time means that each state should have only one arrow coming from it.</p><p>I am to assume that your qualia are axiomatic, yet you expect me to reject it if a computer were to tell me to do the same. This reflects the more general circularity of your argument. You begin at "the rocks in the desert obviously aren't conscious" and conclude that consciousness is more than a computation, but if it where a computation then they would be.</p><p>When you reject the concept of the mind observing itself, that is also a rejection of self awareness which is a critical aspect to being conscious. Moreover there are many systems that can respond to themselves so it is inaccurate. This reminds me of a result from some of the first attempts to mathematically model the mind. The individual neurons being modelled behaved predictably, but when connected together the system began to break down. They form loops which react to themselves, ideas wrenched out of time. This breakdown corresponds quite closely to the way simple systems become undecidable when the paradox of self reference is introduced.</p>

**Comment by joe-antognini** [comment]
Comment written by **Eric** on **06/20/2022 12:59:27**

<p>Fallacy in your argument: there is no such thing as an external observer in this universe.<br>All atoms interacts with each other in this universe. So when Ada is observing a piece of hot iron in a perticular way, she AND the iron collectively forms a consciousness system.</p>

**Comment by joe-antognini** [comment]
Comment written by **Bill Hart** on **06/20/2022 15:40:06**

<p>A computation is more than an organisation of states, but a set of rules for progression between certain states. And even if it is merely an organisation of states, it is clear that the iron bar plus the interpretation of those states is necessary for it to be a computation. In other words, only the combined system could feasibly be conscious if consciousness is a computation.</p>

**Comment by joe-antognini** [comment]
Comment written by **mikkilineni** on **06/20/2022 17:34:51**

<p><a href="https://uploads.disquscdn.com/images/9cbd98509183a9db60253e2236e5b6197be6a06aa876f21da4bbf12bd06adbeb.jpg" rel="nofollow noopener" title="https://uploads.disquscdn.com/images/9cbd98509183a9db60253e2236e5b6197be6a06aa876f21da4bbf12bd06adbeb.jpg">https://uploads.disquscdn.c...</a> <br>I urge the author to read the works of Prof. Mark Burgin and Steven Pinker who have addressed various points he makes in this article.  Our knowledge about matter, energy, information, knowledge, and how they relate to the material world and the mental worlds of biological systems has improved and we now know the limitations of symbolic computing to model the body, brain, and mind. A new science of information processing structures supersedes classical computer science originated with Alan Turing's observation of how humans compute with numbers by "replacing the man in the process of computing a real number by a machine which is capable of only a finite number of conditions.” A Turing machine is a hypothetical device that manipulates symbols on a strip of tape according to a table of rules. Despite its simplicity, a Turing machine can be adapted to simulate the logic of any computer algorithm and is particularly useful in explaining the functions of a CPU inside a computer. In its essence, the Turing machine consists of a repeated algorithm of the “Read-Compute-Write” cycle until the algorithm reaches its end or stops. This simple model combined with John von Neumann’s stored program implementation has become the driver behind the current information processing revolution. However, humans perform many other tasks other than computing with mere numbers.  4E cognition allows information processing tasks that cannot be specified as algorithms.  Elevated cognition allows higher-order inductive modeling, reasoning, and risk mitigation in real-time. The science of information processing structures addresses this.  See <a href="https://tfpis.com/" rel="nofollow noopener" title="https://tfpis.com/">https://tfpis.com/</a><br>Also, I urge the author to read the last paragraph of the last chapter "Computation and its limits" (2012) which points out why general-purpose computers are inadequate to model biological systems which include "self" and the external world in their computations using structures instead of sequences of symbols.</p>

**Comment by joe-antognini** [comment]
Comment written by **antognini** on **06/20/2022 20:02:36**

<p>How would you define a computer in an objective, observer-independent way?</p>

**Comment by joe-antognini** [comment]
Comment written by **antognini** on **06/20/2022 20:05:12**

<p>But if the reason that I am conscious is that my brain is performing a particular computation, who is the other observer that collectively forms a conscious system with my brain?  I seem to be conscious even if there are no other observers in the universe.</p>

**Comment by joe-antognini** [comment]
Comment written by **Mixy Pisa** on **06/20/2022 22:18:38**

<p>Ask Turing.</p>

**Comment by joe-antognini** [comment]
Comment written by **amdson** on **06/21/2022 07:31:36**

<p>Wow, it's uncanny how well this post mirrors my own thoughts, down to the phrase "consciousness is observer independent". Weirdly though, I ended up rejecting the conclusion. Normally I wouldn't expose people to my armchair philosophy, but I'm curious what you'll think, so here goes:</p><p>1. Consciousness, if present, should depend on dynamics, not just state. E.g. frozen snapshots of a conscious being probably shouldn't be conscious themselves.</p><p>2. (Building on 1) Because a full definition of consciousness requires dynamics, a measurement of consciousness can't just give one "conscious" snapshot. It needs to dynamically continue to measure consciousness, which is much stricter than Joe's original requirement.</p><p>3. If measurements of an iron bar are dynamically changing in order to find consciousness in random spins in the iron bar, then picking the correct measurements actually requires immense computation. In fact, it would require simulating a conscious being, while also simulating the dynamics of the iron bar. I'd argue the consciousness you're measuring is then real, but not a property of the iron bar at all.</p><p>- Interestingly, if true, this would be pretty much the same as the reason Maxwell's demon doesn't actually reduce entropy. The entropy required for computation offsets the entropy saved by the demon.</p><p>Joe- I'm honestly so surprised at this post, I feel almost as if you read my mind while writing. Did you have any other ideas about using observer independence? I think there must be some really interesting additional takeaways from that idea, more along the lines of detecting consciousness or reinterpreting it.</p>

**Comment by joe-antognini** [comment]
Comment written by **Johannes Kasimir** on **06/21/2022 08:03:49**

<p>The argument requires that it's actually possible to find a mapping between the state of the iron bar and the state of the Turing machine, under which the  iron bar appears to run consciousness.exe.</p><p>Is this always possible? The author just waves away this question, saying one of Alice's and Bob's friends will find the mapping.</p><p>It is in fact not possible. The number of (bijective) mappings between the systems is much smaller than the number of programs the Turing machine can run.</p><p>Intuitive examples:</p><p>Consider linear functions from R^n to R^n (matrices). Can a similarity transformation transform any matrix into any other matrix? Obviously not. The same is of course also true for general nonlinear functions.</p><p>Imagine two Turing machines running different programs. Is there a mapping between their states (a similarity transformation) such that they seem to run the same program? Not generally!</p><p>Personally I share the feeling that consciousness must be a physical phenomenon. But this argument is unconvincing.</p>

**Comment by joe-antognini** [comment]
Comment written by **antognini** on **06/21/2022 20:31:30**

<p>It is always possible for there to be a mapping between the physical states of a computer and the Turing machine states. The operation of the Turing machine can be completely specified by an encoding of bits. You just need to write down the current entries on the ticker tape, the location of the head, and the state transition table.  All of this can be done with a sequence of bits.</p><p>Once you have encoded your Turing machine as a sequence of bits, you then just need a mapping of physical states to those bits.  So, if I see an atom pointing up in position n, does this correspond to a 0 or a 1 in my encoding?.  If my Turing machine cane be specified by N bits in all, there are 2^N possible encodings.  If the Turing machine then operates for M steps, there are 2^MN possible encodings.  Obviously for any non-trivial program this is an extremely large number, but it is nevertheless finite.  In principle some observer can have an encoding where they observe the random orientation flips in the iron bar to correspond to the operation of a Turing machine they are interested in.</p>

**Comment by joe-antognini** [comment]
Comment written by **antognini** on **06/21/2022 20:36:42**

<p>I agree that computation is not just a static set of states. But the Turing machine just has a table that says, e.g., "If I am in state A and read a 0, transition to state B and move the head to the left."  It says nothing about the physical mechanism by which that progression occurs.</p><p>All we can do is look at the progression of states in the machine and check to make sure that it is following the state transition table of the Turing machine correctly.  If it is, our "Turing machine" is working correctly (however it might actually be operating).  If it violates the rules, then it is not computing correctly.</p><p></p><blockquote>In other words, only the combined system could feasibly be conscious if consciousness is a computation.</blockquote><p></p><p>Yes, this is exactly correct.  And since we know consciousness does not require a "combined system," (no external conscious observer is necessary for me to be conscious), we are forced to conclude that consciousness cannot be reduced to computation.</p>

**Comment by joe-antognini** [comment]
Comment written by **Random Tune** on **07/29/2022 06:02:27**

<p>I skimmed through your article and I think it deserves a patient read and I will come back to it later. Thanks for putting your thoughts about consciousness. I am a software engineer and just have lots of enthusiasm into the philosophy of consciousness.</p><p>Did you do any study or your own thought about this question - did consciousness arise from natural selection as a feature that helped species survive better? When I think about this I conclude that consciousness was not required to achieve the the functionality of our current state (human or other advanced brain animals/creatures). The logic is a robot made of metal and semiconductor with AI can mimic all human like functionality and if we assume computation &lt;&gt; consciousness then we will agree that robot won't be conscious. It would just compute and do the human routines. Fall in love, get angry, feel pain, see colors all in terms of data and logic. Love here will be a data state.</p><p>So the question would then be what purpose consciousness served that it was evolved? Natural things whether physical laws or biological laws (which are physical too) takes shortest and easiest path. If we consider consciousness has some complex mechanism on top of data+logic why evolution didn't stop with data+logic and get us where we are now without the conscious layer.</p><p>Another way of phrasing it would be is consciousness a biproduct of evolution. It just popped up and since it didn't bring negative effect it remained in DNA.</p><p>Thoughts? References?</p>

**Comment by joe-antognini** [comment]
Comment written by **antognini** on **07/29/2022 19:03:54**

<p>I think the basic question you are asking is "can consciousness have a causal impact on an animal's behavior?" If it can, then it's clear why it may have been necessary for it to evolve.  But the alternative, in which consciousness has no causal impact on the physical world, but just observes it (with the illusion of making decisions and acting) is called epiphenomenalism.</p><p>For me, the main difficulty with epiphenomenalism is not the question of why evolution bothered with consciousness.  Maybe it was just an inevitable consequence of neural activity.  The main difficulty for me is why conscious states are so well correlated to survival behavior.  If I eat calorie-rich food, I find that experience pleasurable.  If I break my arm, I find that experience painful.  But if there is no causal connection between consciousness and the physical world, there would be no reason for this to be the case.  I could find breaking my arm pleasurable and eating a donut painful.  But there's always a perfect correlation between my subjective states and survivability.</p>

**Comment by joe-antognini** [comment]
Comment written by **Andrew** on **10/25/2022 20:23:01**

<p>Then every physical system is a computer</p>

**Comment by joe-antognini** [comment]
Comment written by **Mixy Pisa** on **10/25/2022 21:34:53**

<p>No.</p>

**Comment by joe-antognini** [comment]
Comment written by **Mixy Pisa** on **10/25/2022 21:42:22**

<p>That does not follow.</p>

**Comment by jgivoni** [comment]
What happens if you replace "consciousness" with "calculate prime numbers" throughout your argument? Won't it invariably lead to the conclusion that calculating prime numbers is not computation? As far as calculating prime numbers is observer independent, the rest of the argument is identical, as far as I can gather... Please let me know if there is something critical I have missed!
